packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1091,16 +1103,22 @@ def process_payload(self, payload: BaseTxPayload) -> None:
             )
 
         sender = payload.sender
-        if sender not in self.synchronized_data.participants:
+        if sender not in self.accepting_payloads_from:
             raise ABCIAppInternalError(
-                f"{sender} not in list of participants: {sorted(self.synchronized_data.participants)}"
+                f"{sender} not in list of participants: {sorted(self.accepting_payloads_from)}"
             )
 
         if sender in self.collection:
             raise ABCIAppInternalError(
                 f"sender {sender} has already sent value for round: {self.round_id}"
             )
 
+        if self._hash_length:
+            content = payload.data.get(self.payload_attribute)
+            if not content or len(content) % self._hash_length:
+                msg = "FSM design error: expecting serialized data of chunk size"
+                raise ABCIAppInternalError(f"{msg} {self._hash_length}, got: {content}")
+
```

Comment:
this compensates the loss of assertion on the serialized data format in by raising a `ABCIAppInternalError` on processing the payload.

On a separate note: I suggest to move these checks in process_payload method to the check_payload method and call latter from within the former method. I noticed they raise slightly different errors, `ABCIAppInternalError` vs `TransactionNotValidError`; is there a reason?

Reply:
it is now, logging error on processing, and raising the `TransactionNotValidError` captured for logging in the `ABCIRoundHandler.deliver_tx` method